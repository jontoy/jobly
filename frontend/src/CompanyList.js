import React from 'react';
import {Link} from 'react-router-dom';
import CompanyCard from './CompanyCard';

const CompanyList = ({companies}) => {
    return (
        <ul className="CompanyList list-unstyled">
            {companies.map(({description, logo_url, handle, name}, idx) => 
            <li key={idx}>
                <Link to={`/companies/${handle}`}>
                    <CompanyCard 
                        description={description} 
                        logo_url={logo_url} 
                        name={name}/>
                </Link>
            </li>)}
        </ul>
    )
}

export default CompanyList;