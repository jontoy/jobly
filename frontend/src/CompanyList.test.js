import React from 'react';
import { render } from '@testing-library/react';
import CompanyList from './CompanyList';
import { MemoryRouter } from 'react-router-dom';

it('renders without crashing', () => {
    render(<CompanyList companies={[]} />);
});
  
it('matches snapshot', () => {
      const {asFragment} = render(
      <MemoryRouter>
        <CompanyList companies={[]}/>
      </MemoryRouter>
      );
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot with jobs data', () => {
    const companies = [
        {handle:'company-one', description:'test1', name:'Company One'},
        {handle:'company-two', description:'test2', name:'Company Two', logo_url:'https://image.flaticon.com/icons/svg/25/25447.svg'}]
    const {asFragment} = render(
        <MemoryRouter>
         <CompanyList companies={companies}/>
        </MemoryRouter>
      );
    expect(asFragment()).toMatchSnapshot();
});

