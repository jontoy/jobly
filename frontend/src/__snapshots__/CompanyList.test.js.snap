// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="CompanyList list-unstyled"
  />
</DocumentFragment>
`;

exports[`matches snapshot with jobs data 1`] = `
<DocumentFragment>
  <ul
    class="CompanyList list-unstyled"
  >
    <li>
      <a
        href="/companies/company-one"
      >
        <div
          class="CompanyCard card p-2 mb-1"
        >
          <h5
            class="card-title"
          >
            Company One
          </h5>
          <img
            alt="Company One"
            class="img-thumbnail ml-auto"
            src="office_icon.png"
          />
          <p
            class="card-text"
          >
            test1
          </p>
        </div>
      </a>
    </li>
    <li>
      <a
        href="/companies/company-two"
      >
        <div
          class="CompanyCard card p-2 mb-1"
        >
          <h5
            class="card-title"
          >
            Company Two
          </h5>
          <img
            alt="Company Two"
            class="img-thumbnail ml-auto"
            src="https://image.flaticon.com/icons/svg/25/25447.svg"
          />
          <p
            class="card-text"
          >
            test2
          </p>
        </div>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
